hiero-observability/hiero-metrics-core/src/main/java/org/hiero/metrics/api/StatelessMetric.java

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.metrics.api;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.DoubleSupplier;
+import java.util.function.LongSupplier;
+import org.hiero.metrics.api.core.Metric;
+import org.hiero.metrics.api.core.MetricKey;
+import org.hiero.metrics.api.core.MetricType;
+import org.hiero.metrics.api.core.NumberSupplier;
+import org.hiero.metrics.internal.StatelessMetricImpl;
+
+/**
+ * A stateless metric of type {@link MetricType#GAUGE} that doesn't hold any state
+ * and gets/exports its value using provided suppliers.
+ * <p>
+ * Value suppliers are provided per data point during metric construction using
+ * {@link Builder#registerDataPoint(NumberSupplier, String...)}, or could be added later
+ * using {@link #registerDataPoint(NumberSupplier, String...)}.
+ */
+public interface StatelessMetric extends Metric {
+
+    /**
+     * Create a metric key for a {@link StatelessMetric} with the given name. <br>
+     * See {@link org.hiero.metrics.api.utils.MetricUtils#validateNameCharacters(String)} for name requirements.
+     *
+     * @param name the name of the metric
+     * @return the metric key
+     */
+    @NonNull
+    static MetricKey<StatelessMetric> key(@NonNull String name) {
+        return MetricKey.of(name, StatelessMetric.class);
+    }
+
+    /**
+     * Create a builder for a {@link StatelessMetric} with the given metric key.
+     *
+     * @param key the metric key
+     * @return the builder
+     */
+    @NonNull
+    static Builder builder(@NonNull MetricKey<StatelessMetric> key) {
+        return new Builder(key);
+    }
+
+    /**
+     * Create a builder for a {@link StatelessMetric} with the given metric name. <br>
+     * See {@link org.hiero.metrics.api.utils.MetricUtils#validateNameCharacters(String)} for name requirements.
+     *
+     * @param name the metric name
+     * @return the builder
+     */
+    @NonNull
+    static Builder builder(@NonNull String name) {
+        return builder(key(name));
+    }
+
+    /**
+     * Register a data point with the given value supplier and labels.
+     * <p>
+     * Provided label names must match the dynamic labels specified during metric creation.
+     * Constant labels should not be provided here, as they are already associated with the metric.
+     * Order doesn't matter, but for efficiency, it is recommended to provide label names in alphabetical order.
+     *
+     * @param valueSupplier         the supplier to get the numerical value of the data point
+     * @param labelNamesAndValues   alternating label names and values, e.g. "label1", "value1", "label2", "value2"
```

Comment:
A map (String to String) would be more appropriate here. Not sure about performance, though